types/index.d.ts

```diff
@@ -108,6 +108,8 @@ declare module 'mongoose' {
     [k: string]: any
   }
 
+  export type AnyObjectAnyId = AnyObject & { _id?: string | number | boolean | object };
```

Comment:
Why is this necessary? Can't we do `_id?: any` ?

Reply:
`{ _id?: any }` doesn't work; it was what I initially tried. There's a conditional type somewhere that doesn't agree with that. :)

And `AnyObject` should already include `{ _id?: any }` implicitly by how it's defined:
```
  export interface AnyObject {
    [k: string]: any
  }
```

You can try replacing it with `_id?: any`; you can see that the error comes back.